### PR TITLE
Support storage stored access policy and update storage SDK

### PR DESCRIPTION
**Pull request by yaxia**
1. Updated storage SDK to 0.4.1.
2. Supported stored access policy for blob, table and queue services.

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@yaxia__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (yaxia). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, or work for Microsoft Open Technologies, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

File: lib/util/storage.util._js [hunk 1]
```diff
@@ -738,19 +748,265 @@ StorageUtil.isValidRetention = function(value) {
...
+  var policies = [];
+  StorageUtil.startProgress(policySettings.tips);
+  try {
+    policies = StorageUtil.performStorageOperation(policySettings.getAclOperation, _, policySettings.resourceName, policySettings.storageOptions).signedIdentifiers;
+
+    var newPolicy = generatePolicy(policySettings.policyName, policySettings.policyOptions.start, policySettings.policyOptions.expiry, policySettings.policyOptions.permissions);
+    modifyPolicySet(newPolicy, policies, policySettings.accessType, operation);
+
+    if (policySettings.accessType === StorageUtil.AccessType.Container) {
+      StorageUtil.performStorageOperation(policySettings.setAclOperation, _, policySettings.resourceName, policies, null, policySettings.storageOptions);
```
Comment (yugangw-msft) [inline lib/util/storage.util._js]: Just  questiuon that  you do mean to pass the "_" (streamline implict callback) in the middle, rather usually the last argument

File: lib/util/storage.util._js [hunk 2]
```diff
@@ -738,19 +748,265 @@ StorageUtil.isValidRetention = function(value) {
...
+
+/**
+* Modify the sotrage policies set that would be sent
+*/
+function modifyPolicySet(newPolicy, existingPolicies, accessType, operation) {
+  var index = -1;
+  for (var i = 0; i < existingPolicies.length; i++) {
+    normalizePolicy(existingPolicies[i], accessType);
+    if (existingPolicies[i].Id === newPolicy.Id) {
+      index = i;
```
Comment (yugangw-msft) [inline lib/util/storage.util._js]: need to break here?

File: lib/util/storage.util._js [hunk 3]
```diff
@@ -738,19 +748,265 @@ StorageUtil.isValidRetention = function(value) {
...
+    }
+  }
+
+  if (operation == StorageUtil.PolicyOperation.Create) {
```
Comment (yugangw-msft) [inline lib/util/storage.util._js]: nit, use ===.?

File: lib/util/storage.util._js (hunk 1 above)
Comment (yaxia) [inline lib/util/storage.util._js]: This is to follow the parameter pattern of StorageUtil.performStorageOperation(storageOperation, _). "_" is required as second parameter in the function declaration, So the parameters for "storageOperation" are behind "_".

File: lib/util/storage.util._js (hunk 2 above)
Comment (yaxia) [inline lib/util/storage.util._js]: The available index is set only when the "if" clause is true, but normalizePolicy() should be applied to all the items. So we don't break here.
And the maximum policy number is 5, so the redundant checking won't have much impact.

File: lib/util/storage.util._js (hunk 3 above)
Comment (yaxia) [inline lib/util/storage.util._js]: Resolved. Thanks.

**Pull request merged**